File: api/schemas/game.py
"""Game-related Pydantic schemas."""

from pydantic import BaseModel
from typing import Optional, List, Tuple


class GameCreate(BaseModel):
    variant: str = "standard"
    game_id: Optional[str] = None


class MoveRequest(BaseModel):
    from_point: Optional[int] = None
    to_point: Optional[int] = None
    move_type: str  # "normal", "enter", "bear_off"
    die_value: Optional[int] = None


class SetPlayerRequest(BaseModel):
    player: str  # "white" or "black"


class PointData(BaseModel):
    number: int
    white_pieces: int
    black_pieces: int


class BoardState(BaseModel):
    points: List[PointData]
    bar_white: int
    bar_black: int
    borne_off_white: int
    borne_off_black: int
    current_player: Optional[str]
    dice: Optional[Tuple[int, int]]
    game_over: bool
    winner: Optional[str]


class LegalMove(BaseModel):
    move_type: str
    from_point: Optional[int]
    to_point: Optional[int]
    die_value: int


class GameState(BaseModel):
    game_id: str
    variant: str
    board: BoardState
    legal_moves: List[LegalMove]
    can_roll: bool

File: app/src/types/game.ts
// Game-related types

export interface PointData {
  number: number;
  white_pieces: number;
  black_pieces: number;
}

export interface BoardState {
  points: PointData[];
  bar_white: number;
  bar_black: number;
  borne_off_white: number;
  borne_off_black: number;
  current_player: string | null;
  dice: [number, number] | null;
  game_over: boolean;
  winner: string | null;
}

export interface LegalMove {
  move_type: string;
  from_point: number | null;
  to_point: number | null;
  die_value: number;
}

export interface GameState {
  game_id: string;
  variant: string;
  board: BoardState;
  legal_moves: LegalMove[];
  can_roll: boolean;
}

export interface MoveRequest {
  from_point?: number | null;
  to_point?: number | null;
  move_type: string;
  die_value?: number | null;
}

export type GameMode = 'local' | 'ai';
export type AIDifficulty = 'easy' | 'medium' | 'hard';
export type PlayerColor = 'white' | 'black';

export interface MoveAnimation {
  from_point: number;
  to_point: number;
  move_type: string;
}
